### PR TITLE
Add support for Tensorflow SparseTensors: merging layers.

### PR DESCRIPTION
**Pull request by hertschuh**
Added `tf.SparseTensor` support for ops:
- add
- concatenate
- maximum
- minimum
- multiply
- subtract

Added `tf.SparseTensor` support for merging layers:
- Add
- Average
- Concatenate
- Maximum
- Minimum
- Multiply
- Subtract

Note that the `Dot` merging layer will be addressed in a separate PR.

### CODE REVIEW

File: keras_core/layers/merging/add.py [hunk 1]
```diff
@@ -32,7 +33,7 @@ class Add(Merge):
     def _merge_function(self, inputs):
         output = inputs[0]
         for i in range(1, len(inputs)):
-            output = output + inputs[i]
+            output = ops.add(output, inputs[i])
```
Comment (fchollet) [inline keras_core/layers/merging/add.py:36]: Very strange that TF cannot `+` sparse tensors...

File: keras_core/ops/numpy_test.py [hunk 2]
```diff
@@ -2376,6 +2490,81 @@ def test_digitize(self):
...
+    def test_sparse(
+        self,
+        op_function,
+        op_class,
+        np_op,
+        mixed_inputs_produce_sparse_output=False,
+    ):
+        import tensorflow as tf
+
+        x = tf.SparseTensor(
```
Comment (fchollet) [inline keras_core/ops/numpy_test.py:2541]: Please also test execution/correctness for the sparse/dense mixture case, i.e. x sparse and y dense or inversely.

**Review by fchollet — COMMENTED** [review]
Thanks for the PR! 👍

File: keras_core/layers/merging/add.py (hunk 1 above)
Comment (hertschuh) [reply keras_core/layers/merging/add.py:36]: Indeed

```
      3 x = tf.SparseTensor(indices=[[0, 0], [1, 2]], values=[2, 3], dense_shape=(3, 3))
----> 4 x + x

TypeError: unsupported operand type(s) for +: 'SparseTensor' and 'SparseTensor'
```

Only `__div__`, `__mul__`, `__truediv__` are supported: https://www.tensorflow.org/api_docs/python/tf/sparse/SparseTensor

File: keras_core/ops/numpy_test.py (hunk 2 above)
Comment (hertschuh) [reply keras_core/ops/numpy_test.py:2541]: Oh, that's exactly what this test is. `x` and `y` are sparse, `z` is dense, I verify `x + z` (sparse + dense), `z + x` (dense + sparse) and `x + y` (sparse + sparse).

I can add comments to clarify

**Review by hertschuh — COMMENTED** [review]
Thanks for the review!

**Pull request merged**